packages/contracts-bedrock/test/DeployImplementations.t.sol

```diff
@@ -267,13 +275,142 @@ contract DeployImplementations_Test is Test {
+
+        address srOptimismPortalImpl = address(0xe2F826324b2faf99E513D16D266c3F80aE87832B);
+        vm.etch(address(srOptimismPortalImpl), hex"01");
+        deployImplementations.deployOptimismPortalImpl(dii, dio);
+        assertEq(srOptimismPortalImpl, address(dio.optimismPortalImpl()));
+
+        address srDelayedWETHImpl = address(0x71e966Ae981d1ce531a7b6d23DC0f27B38409087);
+        vm.etch(address(srDelayedWETHImpl), hex"01");
+        deployImplementations.deployDelayedWETHImpl(dii, dio);
+        assertEq(srDelayedWETHImpl, address(dio.delayedWETHImpl()));
+
+        address srPreimageOracleSingleton = address(0x9c065e11870B891D214Bc2Da7EF1f9DDFA1BE277);
+        vm.etch(address(srPreimageOracleSingleton), hex"01");
+        deployImplementations.deployPreimageOracleSingleton(dii, dio);
+        assertEq(srPreimageOracleSingleton, address(dio.preimageOracleSingleton()));
+
+        address srMipsSingleton = address(0x16e83cE5Ce29BF90AD9Da06D2fE6a15d5f344ce4);
+        vm.etch(address(srMipsSingleton), hex"01");
+        deployImplementations.deployMipsSingleton(dii, dio);
+        assertEq(srMipsSingleton, address(dio.mipsSingleton()));
+
+        address srDisputeGameFactoryImpl = address(0xc641A33cab81C559F2bd4b21EA34C290E2440C2B);
+        vm.etch(address(srDisputeGameFactoryImpl), hex"01");
+        deployImplementations.deployDisputeGameFactoryImpl(dii, dio);
+        assertEq(srDisputeGameFactoryImpl, address(dio.disputeGameFactoryImpl()));
+    }
+
+    function test_deployAtNonExistentRelease_reverts() public {
+        string memory unknownRelease = "op-contracts/v0.0.0";
+        dii.set(dii.release.selector, unknownRelease);
+
+        bytes memory expectedErr =
+            bytes(string.concat("DeployImplementations: failed to deploy release ", unknownRelease));
+
+        vm.expectRevert(expectedErr);
+        deployImplementations.deploySystemConfigImpl(dii, dio);
+
+        vm.expectRevert(expectedErr);
+        deployImplementations.deployL1CrossDomainMessengerImpl(dii, dio);
+
+        vm.expectRevert(expectedErr);
+        deployImplementations.deployL1ERC721BridgeImpl(dii, dio);
+
+        vm.expectRevert(expectedErr);
+        deployImplementations.deployL1StandardBridgeImpl(dii, dio);
+
+        vm.expectRevert(expectedErr);
+        deployImplementations.deployOptimismMintableERC20FactoryImpl(dii, dio);
+
+        // TODO: Uncomment the code below when OPContractsManager is deployed based on release. Superchain-registry
+        // doesn't contain OPContractsManager yet.
+        // dii.set(dii.superchainConfigProxy.selector, address(superchainConfigProxy));
+        // dii.set(dii.protocolVersionsProxy.selector, address(protocolVersionsProxy));
+        // vm.etch(address(superchainConfigProxy), hex"01");
+        // vm.etch(address(protocolVersionsProxy), hex"01");
+        // vm.expectRevert(expectedErr);
+        // deployImplementations.deployOPContractsManagerImpl(dii, dio);
```

Comment:
Not quite following when we'd uncomment this

Reply:
Should Op contracts manager implementation be included in the superchain registry at some stage? https://github.com/ethereum-optimism/superchain-registry/blob/main/validation/standard/standard-versions.toml

If so, this code would be uncommented. Right now, in `DeployImplementation.s.sol` I'm deploying Op contracts manager impl fresh everytime.

Reply:
Hmm good question, will need to think on that. I can see arguments for both. For now I like this TODO, and I added a sticky to the figjam to track this